src/generators/swagger2.ts

```diff
@@ -0,0 +1,26 @@
+import { CodeGenOptions } from "../options/options";
+import { transformToCodeWithMustache } from "../transform/transformToCodeWithMustache";
+import { getViewForSwagger2, ViewData } from "../getViewForSwagger2";
+
+interface CodeGenerator {
+  getViewData(opts: CodeGenOptions): ViewData;
+  getCode(opts: CodeGenOptions): string;
+}
+
+function verifyThatWeAreGeneratingForSwagger2(opts: CodeGenOptions): void {
+  if (opts.swagger.swagger !== "2.0") {
+    throw new Error("Only Swagger 2 specs are supported");
+  }
+}
+
+export const Swagger2Gen: CodeGenerator = {
+  getViewData: function(opts: CodeGenOptions): ViewData {
+    verifyThatWeAreGeneratingForSwagger2(opts);
+
+    return getViewForSwagger2(opts);
+  },
+  getCode: function(opts: CodeGenOptions): string {
+    const data = this.getViewData(opts);
+    return transformToCodeWithMustache(data, opts.template, opts.mustache);
+  }
+};
```

Comment:
Hm, the Swagger2Gen wrapper is a bit redundant, and might be prettier if we simply export the functions directly in the module instead of wrapping it in the Swagger2Gen wrapper?

Ie:
````
export function getViewData(opts: CodeGenOptions): ViewData {
    verifyThatWeAreGeneratingForSwagger2(opts);

    return getViewForSwagger2(opts);
};

export function getCode(opts: CodeGenOptions): string {
    const data = this.getViewData(opts);
    return transformToCodeWithMustache(data, opts.template, opts.mustache);
}
````

Reply:
I was thinking about making it more abstract for to make it easier to implement different version like Swagger 3.0 and OpenAPI. #106

Reply:
Hm, that is a good point, the wrapper definitely helps then. Let's keep it as is then

Reply:
I moved it to a new file instead, hopefully that helps understanding it a bit more.